server/src/main/java/org/elasticsearch/common/network/CloseableChannel.java

```diff
@@ -49,50 +47,18 @@
     boolean isOpen();
 
     /**
-     * Closes the channel without blocking.
-     *
-     * @param channel to close
-     */
-    static <C extends CloseableChannel> void closeChannel(C channel) {
-        closeChannel(channel, false);
-    }
-
-    /**
-     * Closes the channel.
-     *
-     * @param channel  to close
-     * @param blocking indicates if we should block on channel close
-     */
-    static <C extends CloseableChannel> void closeChannel(C channel, boolean blocking) {
-        closeChannels(Collections.singletonList(channel), blocking);
-    }
-
-    /**
-     * Closes the channels.
+     * Closes the channels in a blocking manner.
      *
      * @param channels to close
-     * @param blocking indicates if we should block on channel close
      */
-    static <C extends CloseableChannel> void closeChannels(List<C> channels, boolean blocking) {
-        try {
-            IOUtils.close(channels);
-        } catch (IOException e) {
-            // The CloseableChannel#close method does not throw IOException, so this should not occur.
-            throw new AssertionError(e);
-        }
-        if (blocking) {
-            ArrayList<ActionFuture<Void>> futures = new ArrayList<>(channels.size());
+    static <C extends CloseableChannel> void closeChannelsBlocking(List<C> channels) {
+        if (channels.isEmpty() == false) {
+            Releasables.close(channels);
+            final PlainActionFuture<Collection<Void>> future = PlainActionFuture.newFuture();
+            final ActionListener<Void> closeListener = new GroupedActionListener<>(future, channels.size());
```

Comment:
Just an aesthetic fix to use `GroupedActionListener` here, it seemed weird to loop over blocking `.get` and catch+re-apply interrupts in a loop.